libs/updateOrCreate.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * The goal of this function is create or update the given entries according to if they already
+ * exist in the cozy or not
+ *
+ * - `entries` is an array of objects with any attributes :
+ *
+ * - `doctype` (string) is the cozy doctype where the entries should be saved
+ *
+ * - `filters` (array) is the list of attributes in each entry should be used to check if an entry
+ *   is already saved in the cozy
+ *
+ * @module updateOrCreate
+ */
+const bluebird = require('bluebird')
+const log = require('./logger').namespace('updateOrCreate')
+const cozy = require('./cozyclient')
+
+module.exports = (entries = [], doctype, filters = []) => {
```

Comment:
I would rename `filters` into `idAttributes` since to me filters would be a function.

Reply:
Good point, at first I thank it was a function too. Also `mathingAttributes` could to the job (to match my other suggestion below. :)